File: docs/snippets/init_session_eth_key.rb
require 'eth'
require 'jwt'


key       = Eth::Key.new(priv: ENV['ETH_PRIVATE_KEY'])
payload   = { address: key.address,
              callback_url: '<YOUR CALLBACK URL>' }
signature = key.personal_sign(payload.to_json)
token     = JWT.encode(payload.merge(signature: signature), nil, 'none')

File: docs/snippets/init_session_eth_key.js
const jwt = require('jsonwebtoken')
const Accounts = require('web3-eth-accounts')

const account = (new Accounts()).privateKeyToAccount(process.env.ETH_PRIVATE_KEY)
const payload = { address: account.address, callback_url: '<YOUR CALLBACK URL>' }
const signature = account.sign(JSON.stringify(payload))
const token = jwt.sign({ ...payload, signature }, null, { algorithm: 'none' })
